### ISSUE TITLE
array name could be a constant

### ISSUE THREAD

**Issue by x87** [sannybuilder/dev#250]
currently this code does not compile:

```
const arr = 1@
arr(0@,1i) = 0
```

this code, however, does:

```
int arr[1]
arr[0@] = 0
```

https://github.com/sannybuilder/dev/wiki/Grammar#arrays

**Comment by x87** [comment]
also works with opcode
```
0004: arr(0@,1i) = 0
```

**Comment by x87** [comment]
does not work with float and string arrays

```
const arr = 1@
arr(0@,1f) = 0.0
```

```
const arr = 1@
arr(0@,1s) = ''
```

**Comment by x87** [comment]
fixed in 4.0.0
